bats/tests/k8s/verify-cached-images.bats

```diff
@@ -0,0 +1,59 @@
+# Test cases 57, 58
+
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start rancher desktop' {
+    # Start first so we can use `rdctl set` in all the calls that change the k8s version
+    start_kubernetes
+    wait_for_apiserver
+    # the docker context "rancher-desktop" may not have been written
+    # even though the apiserver is already running
+    wait_for_container_engine
+}
+
+test_k8s_version_has_correct_cached_extension() {
+    local K8S_VERSION=$1
+    local EXTENSION=$2
+    rdctl set --kubernetes-version "$K8S_VERSION"
+    wait_for_apiserver "$K8S_VERSION"
+    wait_for_container_engine
+    run ls "$PATH_CACHE/k3s/v${K8S_VERSION}"*k3s*/k3s-airgap-images-amd64."${EXTENSION}"
+    assert_success
+}
+
+@test 'verify k8s 1.23.13 uses .tar.zst in the cache' {
+    test_k8s_version_has_correct_cached_extension "1.23.13" "tar.zst"
+}
+
+@test 'verify k8s 1.25.3 uses .tar.zst in the cache' {
+    test_k8s_version_has_correct_cached_extension "1.25.3" "tar.zst"
+}
+
+@test 'verify k8s 1.18.17 uses .tar in the cache' {
+    test_k8s_version_has_correct_cached_extension "1.18.17" "tar"
+}
+
+@test 'verify k8s 1.26.3 uses .tar.zst in the cache' {
+    test_k8s_version_has_correct_cached_extension "1.26.3" "tar.zst"
+}
+
+# Linux run didn't shutdown without this test.
```

Comment:
Not sure what this mean; BATS will not shutdown RD unless the test does it explicitly; this is not specific to Linux.

I'm not sure if we need to shut down RD at the end of a test. Sometimes it is useful if the VM is still running after a failed test to investigate, so I think I prefer to not shut down at the end of each test file.

Each file starts with a factory-reset anyways.

Anyways, we can remove the shutdown globally across all tests later, but the comment seems misleading and should be fixed or removed.

Reply:
I just read a comment in another test that bats on Linux doesn't exit unless you shut down RD first. So I guess for now we do need to add the shutdown logic to each test file. But shouldn't it go into a `teardown` function instead?